app/config.go

```diff
@@ -54,56 +47,33 @@ func (s *Server) UpdateConfig(f func(*model.Config)) {
 	old := s.Config()
 	updated := old.Clone()
 	f(updated)
-	s.config.Store(updated)
-
-	s.InvokeConfigListeners(old, updated)
+	if _, err := s.configStore.Set(updated); err != nil {
+		mlog.Error("Failed to update config", mlog.Err(err))
+	}
 }
 
 func (a *App) UpdateConfig(f func(*model.Config)) {
 	a.Srv.UpdateConfig(f)
 }
 
 func (a *App) PersistConfig() {
-	config.SaveConfig(a.ConfigFileName(), a.Config())
-}
-
-func (s *Server) LoadConfig(configFile string) *model.AppError {
```

Comment:
The concept of `LoadConfig` doesn't make sense in a world where we can source from non-files, so this functionality has been eliminated altogether.